shared/src/main/scala/scorex/crypto/authds/avltree/batch/BatchAVLProver.scala

```diff
@@ -471,8 +471,8 @@ class BatchAVLProver[D <: Digest, HF <: CryptographicHash[D]](val keyLength: Int
     }
 
     val (minTree, maxTree, treeHeight) = checkTreeHelper(topNode)
-    require(minTree.key sameElements NegativeInfinityKey)
-    require(maxTree.nextLeafKey sameElements PositiveInfinityKey)
+    require(minTree.key.value sameElements NegativeInfinityKey.value)
+    require(maxTree.nextLeafKey.value sameElements PositiveInfinityKey.value)
```

Comment:
These changes shouldn't be necessary, as minTree.key and NegativeInfinityKey are case classes.